src/ocf_ctx.c

```diff
@@ -230,6 +230,7 @@ void ocf_ctx_put(ocf_ctx_t ctx)
 	env_rmutex_lock(&ctx->lock);
 	ENV_BUG_ON(!list_empty(&ctx->caches));
 	env_rmutex_unlock(&ctx->lock);
+	env_rmutex_destroy(&ctx->lock);
```

Comment:
This lock seems to destroyed too early causing code to break. That's why functional tests are failing.

Reply:
thank you, all checks have passed. but please allow me to add few more commits before merging.